logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/featureConfig/GetAndSaveFileSharingStatusUseCase.kt

```diff
@@ -0,0 +1,52 @@
+package com.wire.kalium.logic.feature.featureConfig
+
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.configuration.UserConfigRepository
+import com.wire.kalium.logic.data.featureConfig.FeatureConfigRepository
+import com.wire.kalium.logic.data.featureConfig.FileSharingModel
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.network.exceptions.KaliumException
+import com.wire.kalium.network.exceptions.isNoTeam
+
+interface GetAndSaveFileSharingStatusUseCase {
+    suspend operator fun invoke(): GetFileSharingStatusResult
+}
+
+class GetFileSharingStatusUseCaseImpl(
+    private val userConfigRepository: UserConfigRepository,
+    private val featureConfigRepository: FeatureConfigRepository
+) : GetAndSaveFileSharingStatusUseCase {
+    override suspend operator fun invoke(): GetFileSharingStatusResult =
+        featureConfigRepository.getFileSharingFeatureConfig().fold({
+            if (
+                it is NetworkFailure.ServerMiscommunication &&
+                it.kaliumException is KaliumException.InvalidRequestError
+            ) {
+                if (it.kaliumException.isNoTeam()) {
+                    GetFileSharingStatusResult.Failure.NoTeam
+                } else {
+                    GetFileSharingStatusResult.Failure.OperationDenied
+                }
+            } else {
+                GetFileSharingStatusResult.Failure.Generic(it)
+            }
```

Comment:
What do you think of extracting this block of code to a private function? In my opinion, that way would make it a bit more readable. Not big of a change, but just to understand it better at first. Something like: `mapFileSharingFailure` ? Let me know what you think =)